src/commands/operator/index.ts

```diff
@@ -581,47 +474,58 @@ export default class Operator extends Command {
    * Execute the job
    */
   async executeJob(jobHash: string, tags: (string | number)[]): Promise<boolean> {
-    const job: OperatorJob = this.operatorJobs[jobHash]
-    const network: string = job.network
-    this.networkMonitor.walletNonces[network] = await this.networkMonitor.wallets[network].getTransactionCount()
-    let operate = this.operatorMode === OperatorMode.auto
-    if (this.operatorMode === OperatorMode.manual) {
-      const operatorPrompt: any = await inquirer.prompt([
-        {
-          name: 'shouldContinue',
-          message: `A job is available for execution, would you like to operate?\n`,
-          type: 'confirm',
-          default: true,
-        },
-      ])
-      operate = operatorPrompt.shouldContinue
-    }
+    // quickly check that job is still valid
+    await this.checkJobStatus(jobHash)
+    if (jobHash in this.operatorJobs) {
+      const job: OperatorJob = this.operatorJobs[jobHash]
+      const network: string = job.network
+      let operate = this.operatorMode === OperatorMode.auto
+      if (this.operatorMode === OperatorMode.manual) {
+        const operatorPrompt: any = await inquirer.prompt([
+          {
+            name: 'shouldContinue',
+            message: `A job is available for execution, would you like to operate?\n`,
+            type: 'confirm',
+            default: true,
+          },
+        ])
+        operate = operatorPrompt.shouldContinue
+      }
 
-    if (operate) {
-      const gasPricing: GasPricing = this.networkMonitor.gasPrices[network]
-      const gasPrice: BigNumber = gasPricing.isEip1559 ? gasPricing.maxFeePerGas! : gasPricing.gasPrice!
-
-      const receipt: TransactionReceipt | null = await this.networkMonitor.executeTransaction({
-        network,
-        tags,
-        contract: this.networkMonitor.operatorContract,
-        methodName: 'executeJob',
-        args: [job.payload],
-        gasPrice: gasPrice,
-        gasLimit: job.gasLimit.mul(BigNumber.from('2')),
-      })
-      return receipt !== null
+      if (operate) {
+        const gasPricing: GasPricing = this.networkMonitor.gasPrices[network]
+        const gasPrice: BigNumber = gasPricing.isEip1559 ? gasPricing.maxFeePerGas! : gasPricing.gasPrice!
+
+        const receipt: TransactionReceipt | null = await this.networkMonitor.executeTransaction({
+          network,
+          tags,
+          contract: this.networkMonitor.operatorContract,
```

Comment:
not required, but should we update this to use the operator service going forward?

Reply:
Left it like this for now to not mess up the operator service while I tweak the execute transaction parameters to find the perfect balance.

Reply:
cool